pyiron_base/job/core.py

```diff
@@ -857,18 +857,28 @@ def __del__(self):
 
     def __getitem__(self, item):
         """
-        Get/read data from the HDF5 file or access log files.
+        Get/read data from the HDF5 file, child jobs or access log files.
 
         If the job is :method:`~.decompress`ed, item can also be a file name to
         access the raw output file of that name of the job.  See available file
         with :method:`~.list_files()`.
 
+        `item` is first looked up in this jobs HDF5 file, then in the HDF5 files of any child jobs and finally it is
+        matched against any files in the job directory as described above.
+
         Args:
             item (str, slice): path to the data or key of the data object
 
         Returns:
-            dict, list, float, int: data or data object
+            dict, list, float, int, None: data or data object; if nothing is found None is returned
         """
+        try:
```

Comment:
I agree, I don't like returning `None` for non-existent keys, but I also don't want to refactor this, since presumably a lot of other code depends on this.  So it is what it is.  I can put your suggestion in though, since it is cleaner.